extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapScreenInstaller.kt

```diff
@@ -0,0 +1,98 @@
+package com.mapbox.maps.extension.androidauto
+
+import androidx.car.app.Screen
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * When attaching the many possible experiences onto [MapboxCarMap], this installer allows you to
+ * remove all the boilerplate. Use this installer for creating experiences that will be
+ * attached as long as a screen is in use. To assign an experience to the entire app lifecycle
+ * please use the [MapboxCarMapSessionInstaller].
+ *
+ * Use this to setup each [Screen]. Use the [Screen.mapboxMapInstaller] extension!
+ *
+ * @param screen Android Auto Screen.
+ * @param mapboxCarMap instance that can be also be found from the [install] function.
+ */
+@MapboxExperimental
+class MapboxCarMapScreenInstaller(
+  private val screen: Screen,
+  private val mapboxCarMap: MapboxCarMap
+) {
+  private val created = mutableSetOf<MapboxCarMapObserver>()
+  private val started = mutableSetOf<MapboxCarMapObserver>()
+  private val resumed = mutableSetOf<MapboxCarMapObserver>()
```

Comment:
we could use CopyOnWriteSet instead of mutable set for thread safety?

Reply:
These are builder classes, so I don't think we should add thread safety. You build them once during initialization, should never need to build them with multiple threads